corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -820,11 +820,11 @@ def add_processing_doc_id(self, doc_id):
 
     def remove_processing_doc_id(self, doc_id):
         client = get_redis_connection()
-        client.lrem(self.get_cached_list_key(), 1, doc_id)
+        gevent.spawn(client.lrem, self.get_cached_list_key(), 1, doc_id)
```

Comment:
why is this done in a separate greenlet?

Reply:
Mainly since it's a bit of IO time that I thought I could save. Also, lrem is one of the longer redis functions (O(N)).

It also didn't need to be tracked since removing a doc_id can be done when it's available rather than needing to be done synchronously, since if there is an extra id on resume, that's fine.

Reply:
https://github.com/dimagi/commcare-hq/pull/21007#discussion_r197805792